blockbuster/abci.go

```diff
@@ -0,0 +1,70 @@
+package blockbuster
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+
+	abci "github.com/cometbft/cometbft/abci/types"
+	"github.com/cometbft/cometbft/libs/log"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+type ProposalHandler struct {
+	logger  log.Logger
+	mempool Mempool
+}
+
+func NewProposalHandler(logger log.Logger, mempool Mempool) *ProposalHandler {
+	return &ProposalHandler{
+		logger:  logger,
+		mempool: mempool,
+	}
+}
+
+func (h *ProposalHandler) PrepareProposalHandler() sdk.PrepareProposalHandler {
+	return func(ctx sdk.Context, req abci.RequestPrepareProposal) abci.ResponsePrepareProposal {
+		var (
+			selectedTxs  = make(map[string][]byte)
+			totalTxBytes int64
+		)
+
+		for _, l := range h.mempool.registry {
+			if totalTxBytes < req.MaxTxBytes {
+				laneTxs, err := l.PrepareLane(ctx, req.MaxTxBytes, selectedTxs)
+				if err != nil {
```

Comment:
So in the case of errors in a `PrepareProposal` handler, we'll return empty txs, so when `ProcessProposal` comes, it can decide to error or just process an empty block.